sqlglot/dialects/exasol.py

```diff
@@ -127,6 +127,54 @@ def _timestamp_trunc_sql(self: Exasol.Generator, expression: exp.DateTrunc) -> s
     return _trunc_sql(self, "TIMESTAMP", expression)
 
 
+def is_case_insensitive(node: exp.Expression):
+    return isinstance(node, exp.Collate) and node.text("expression").upper() == "UTF8_LCASE"
+
+
+def _substring_index_sql(self: Exasol.Generator, expression: exp.SubstringIndex) -> str:
+    def build_instr_operands(
+        haystack_node: exp.Expression, delimiter_node: exp.Expression
+    ) -> tuple[str, str, str, str]:
+        haystack_sql = self.sql(haystack_node)
+        delimiter_sql = self.sql(delimiter_node)
+        haystack_search_expr = (
+            f"LOWER({haystack_sql})" if is_case_insensitive(haystack_node) else haystack_sql
+        )
+        delimiter_search_expr = (
+            f"LOWER({delimiter_sql})" if is_case_insensitive(delimiter_node) else delimiter_sql
+        )
+        return haystack_search_expr, delimiter_search_expr, haystack_sql, delimiter_sql
+
+    hay_node = expression.this
+    delim_node = expression.args["delimiter"]
+
+    haystack_search_expr, delimiter_search_expr, haystack_sql, delimiter_sql = build_instr_operands(
+        hay_node, delim_node
+    )
```

Comment:
We don't really need `build_instr_operands`, right? It adds more complexity than solves issues afaict, we can just inline everything:


```suggestion
  haystack_sql = self.sql(expression, 'this')
  delimiter_sql = self.sql(expression, 'delimiter')
```

Also `haystack_search_expr` and `delimiter_search_expr` are used once, we can inline the `LOWER()` logic directly

Reply:
Done